alf/layers.py

```diff
@@ -1025,6 +1026,366 @@ def bias(self):
+            use_bias (bool): whether use bias
+        """
+        super(ParamFC, self).__init__()
+
+        self._input_size = input_size
+        self._output_size = output_size
+        self._activation = activation
+        self._use_bias = use_bias
+        self._kernel_initializer = kernel_initializer
+        self._kernel_init_gain = kernel_init_gain
+        self._bias_init_value = bias_init_value
+
+        self._weight_length = output_size * input_size
+        self.set_weight(torch.randn(1, self._weight_length))
+        if use_bias:
+            self._bias_length = output_size
+            self.set_bias(torch.randn(1, self._bias_length))
+        else:
+            self._bias_length = 0
+            self._bias = None
+
+    @property
+    def weight(self):
+        """Get stored weight tensor or batch of weight tensors."""
+        return self._weight
+
+    @property
+    def bias(self):
+        """Get stored bias tensor or batch of bias tensors."""
+        return self._bias
+
+    @property
+    def weight_length(self):
+        """Get the n_element of a single weight tensor. """
+        return self._weight_length
+
+    @property
+    def bias_length(self):
+        """Get the n_element of a single bias tensor. """
+        return self._bias_length
+
+    def set_weight(self, weight, reinitialize=False):
+        """Store a weight tensor or batch of weight tensors."""
+        assert (weight.ndim == 2 and weight.shape[1] == self._weight_length), (
+            "Input weight has wrong shape %s. Expecting shape (n, %d)" %
+            (weight.shape, self._weight_length))
+        if weight.shape[0] == 1:
+            # non-parallel weight
+            self._groups = 1
+        elif weight.ndim == 2:
+            # parallel weight
+            self._groups = weight.shape[0]
+        weight = weight.view(self._groups, self._output_size, self._input_size)
+        if reinitialize:
+            for i in range(self._groups):
+                if self._kernel_initializer is None:
+                    variance_scaling_init(
+                        weight[i],
+                        gain=self._kernel_init_gain,
+                        nonlinearity=self._activation)
+                else:
+                    self._kernel_initializer(weight[i])
+
+        self._weight = weight
+
+    def set_bias(self, bias, reinitialize=False):
+        """Store a bias tensor or batch of bias tensors."""
+        assert (bias.ndim == 2 and bias.shape[1] == self._bias_length), (
+            "Input bias has wrong shape %s. Expecting shape (n, %d)" %
+            (bias.shape, self._bias_length))
+        if self._groups == 1:
+            # non-parallel bias
+            assert bias.shape[0] == 1, (
+                "Input bias has wrong shape %s. Expecting shape (%d, %d)" %
+                (bias.shape, 1, self.bias_length))
+        else:
+            # parallel weight
+            assert (bias.ndim == 2 and bias.shape[0] == self._groups), (
```

Comment:
bias.ndim==2 is already asserted at line 1115

Reply:
corrected.